common/src/stack/pylib/stack/kvm.py

```diff
@@ -0,0 +1,350 @@
+# @copyright@ # Copyright (c) 2006 - 2019 Teradata
+# All rights reserved. Stacki(r) v5.x stacki.com
+# https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+# @copyright@
+#
+
+import libvirt
+import jinja2
+import tarfile
+import math
+from libvirt import libvirtError
+from stack.util import _exec
+from pathlib import Path
+from collections import defaultdict
+
+# Libvirt has a weird quirk where it prints
+# error messages even if they are handled via
+# exceptions, code taken from:
+# stackoverflow.com/questions/45541725/avoiding-console-prints-by-libvirt-qemu-python-apis
+def libvirt_callback(userdata, err):
+	pass
+
+class VmException(Exception):
+	pass
+
+class Hypervisor:
+
+	"""
+	Class to interface to a kvm hypervisor via
+	the libvirt bindings. Can create and remove vm's,
+	get the current state of guests, and manage
+	storage pools or images.
+	"""
+
```

Comment:
Right now the report command doesn't invoke the hypervisor class now at all (as it makes no active connections to the hypervisor), I think it would make more sense to keep the the config location variable in the report class, as that is the only place it is used for now.